src/core/routing/transfer-routes/multi-transfer.ts

```
@@ -0,0 +1,415 @@
+        
+        // Determine if we should allow this transfer even if already visited
+        let allowTransfer = !alreadyVisited;
+        
+        // Special handling for major interchange points
+        if (alreadyVisited && isMajorInterchange) {
+          // Get this interchange info
+          const interchangeInfo = graph.getMajorInterchangeInfo(stationId);
+          
+          // Allow transfer if both lines are defined for this interchange
+          if (interchangeInfo && 
+              interchangeInfo.lines.includes(lineId) && 
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)